modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,361 @@
+	border: 1px solid #<?php echo sanitize_hex_color_no_hash( $colors['border'] ); ?>;
+	border-top: 0;
+	color: #<?php echo sanitize_hex_color_no_hash( $colors['text'] ); ?>;
+	padding-bottom: 1em;
+}
+.milestone-message {
+	padding-top: 1em
+}
+</style>
+<?php
+	}
+
+	/**
+	 * Localize Front-end Script.
+	 *
+	 * Print the javascript configuration array only if the
+	 * current template has an instance of the widget that
+	 * is still counting down. In all other cases, this
+	 * function will dequeue milestone.js.
+	 *
+	 * Hooks into the "wp_footer" action.
+	 */
+	function localize_script() {
+		if ( empty( self::$config_js['instances'] ) ) {
+			wp_dequeue_script( 'milestone' );
+			return;
+		}
+
+		wp_localize_script( 'milestone', 'MilestoneConfig', array(
+			'instances'         => self::$config_js['instances'],
+			'labels'            => self::get_interval_labels(),
+			'MINUTE_IN_SECONDS' => MINUTE_IN_SECONDS,
+			'HOUR_IN_SECONDS'   => HOUR_IN_SECONDS,
+			'DAY_IN_SECONDS'    => DAY_IN_SECONDS,
+			'WEEK_IN_SECONDS'   => WEEK_IN_SECONDS,
+			'MONTH_IN_SECONDS'  => MONTH_IN_SECONDS,
+			'YEAR_IN_SECONDS'   => YEAR_IN_SECONDS,
+		) );
+	}
+
+	/**
+	 * Widget
+	 */
+	function widget( $args, $instance ) {
+		$instance = $this->sanitize_instance( $instance );
+
+		$milestone = mktime( $instance['hour'], $instance['min'], 0, $instance['month'], $instance['day'], $instance['year'] );
+		$now  = (int) current_time( 'timestamp' );
+		$diff = (int) floor( $milestone - $now );
+
+		$number = 0;
+		$label  = '';
+		$interval_labels = self::get_interval_labels();
+
+		if ( 2 * YEAR_IN_SECONDS <= $diff ) { // more than 2 years - show in years, one decimal point
+			$number = round( $diff / YEAR_IN_SECONDS, 1 );
+			$label  = $interval_labels['years'];
+		} else if ( 3 * MONTH_IN_SECONDS <= $diff ) { // fewer than 2 years - show in months
+			$number = floor( $diff / MONTH_IN_SECONDS );
+			$label  = ( 1 == $number ) ? $interval_labels['month'] : $interval_labels['months'];
+		} else if ( DAY_IN_SECONDS <= $diff ) { // fewer than 3 months - show in days
+			$number = floor( $diff / DAY_IN_SECONDS ) + 1;
+			$label  = ( 1 == $number ) ? $interval_labels['day'] : $interval_labels['days'];
+		} else if ( HOUR_IN_SECONDS <= $diff ) { // less than 1 day - show in hours
+			$number = floor( $diff / HOUR_IN_SECONDS );
+			$label  = ( 1 == $number ) ? $interval_labels['hour'] : $interval_labels['hours'];
+		} else if ( MINUTE_IN_SECONDS <= $diff ) { // less than 1 hour - show in minutes
+			$number = floor( $diff / MINUTE_IN_SECONDS ) + 1;
+			$label = ( 1 == $number ) ? $interval_labels['minute'] : $interval_labels['minutes'];
+		} else { // less than 1 minute - show in seconds
+			$number = $diff;
+			$label = ( 1 == $number ) ? $interval_labels['second'] : $interval_labels['seconds'] ;
+		}
+
+		echo $args['before_widget'];
+
+		$title = apply_filters( 'widget_title', $instance['title'] );
```

Comment:
Could you add a docblock here, something like `/** This filter is documented in core/src/wp-includes/default-widgets.php */`, to make the parser happy?